# Open notebook in Google Colab

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/arsalmairaj2k/taxi_trip_analysis/blob/main/Task1.ipynb)

# Task 1: Dataset Selection & Understanding

## Load Dataset
We load the dataset from our GitHub repository, where it resides.

In [8]:
import pandas as pd

GITHUB_URL = "https://github.com/arsalmairaj2k/taxi_trip_analysis/raw/main/TaxiTripData.xlsx"
df = pd.read_excel(GITHUB_URL)

## Basic Data Exploration
Since we loaded our data from our GitHub repository, we can make sure that our dataset was correctly fetched by applying some basic opertaions on it.

In [10]:
# Display first few rows
df.head()

VendorID tpep_pickup_datetime tpep_dropoff_datetime  passenger_count  \
0       1.0  2020-01-01 00:28:00   2020-01-01 00:33:00              1.0   
1       1.0  2020-01-01 00:35:00   2020-01-01 00:43:00              1.0   
2       1.0  2020-01-01 00:47:00   2020-01-01 00:53:00              1.0   
3       1.0  2020-01-01 00:55:00   2020-01-01 01:00:00              1.0   
4       2.0  2020-01-01 00:01:00   2020-01-01 00:04:00              1.0   

   trip_distance  RatecodeID store_and_fwd_flag  PULocationID  DOLocationID  \
0            1.2         1.0                  N           238           239   
1            1.2         1.0                  N           239           238   
2            0.6         1.0                  N           238           238   
3            0.8         1.0                  N           238           151   
4            0.0         1.0                  N           193           193   

   payment_type  fare_amount  extra  mta_tax  tip_amount  tolls_amount  \
0           1.0          6.0    3.0      0.5        1.47           0.0   
1           1.0          7.0    3.0      0.5        1.50           0.0   
2           1.0          6.0    3.0      0.5        1.00           0.0   
3           1.0          5.5    0.5      0.5        1.36           0.0   
4           2.0          3.5    0.5      0.5        0.00           0.0   

   improvement_surcharge  total_amount  congestion_surcharge  
0                    0.3         11.27                   2.5  
1                    0.3         12.30                   2.5  
2                    0.3         10.80                   2.5  
3                    0.3          8.16                   0.0  
4                    0.3          4.80                   0.0

In [12]:
# Check data types and missing values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1048575 entries, 0 to 1048574
Data columns (total 18 columns):
 #   Column                 Non-Null Count    Dtype  
---  ------                 --------------    -----  
 0   VendorID               1046444 non-null  float64
 1   tpep_pickup_datetime   1048575 non-null  object 
 2   tpep_dropoff_datetime  1048575 non-null  object 
 3   passenger_count        1046444 non-null  float64
 4   trip_distance          1048575 non-null  float64
 5   RatecodeID             1046444 non-null  float64
 6   store_and_fwd_flag     1046444 non-null  object 
 7   PULocationID           1048575 non-null  int64  
 8   DOLocationID           1048575 non-null  int64  
 9   payment_type           1046444 non-null  float64
 10  fare_amount            1048575 non-null  float64
 11  extra                  1048575 non-null  float64
 12  mta_tax                1048575 non-null  float64
 13  tip_amount             1048575 non-null  float64
 14  tolls_amount      

In [14]:
# Summary statistics
df.describe()

VendorID  passenger_count  trip_distance    RatecodeID  \
count  1.046444e+06     1.046444e+06   1.048575e+06  1.046444e+06   
mean   1.678677e+00     1.583992e+00   3.145816e+00  1.075027e+00   
std    4.669849e-01     1.189790e+00   4.169883e+00  9.042037e-01   
min    1.000000e+00     0.000000e+00  -2.218000e+01  1.000000e+00   
25%    1.000000e+00     1.000000e+00   1.000000e+00  1.000000e+00   
50%    2.000000e+00     1.000000e+00   1.680000e+00  1.000000e+00   
75%    2.000000e+00     2.000000e+00   3.190000e+00  1.000000e+00   
max    2.000000e+00     9.000000e+00   2.592200e+02  9.900000e+01   

       PULocationID  DOLocationID  payment_type   fare_amount         extra  \
count  1.048575e+06  1.048575e+06  1.046444e+06  1.048575e+06  1.048575e+06   
mean   1.620965e+02  1.601928e+02  1.328451e+00  1.298749e+01  1.030286e+00   
std    6.625240e+01  7.114689e+01  5.011478e-01  1.307701e+01  1.228418e+00   
min    1.000000e+00  1.000000e+00  1.000000e+00 -1.238000e+03 -7.000000e+00   
25%    1.140000e+02  1.070000e+02  1.000000e+00  6.000000e+00  0.000000e+00   
50%    1.610000e+02  1.610000e+02  1.000000e+00  9.000000e+00  5.000000e-01   
75%    2.330000e+02  2.330000e+02  2.000000e+00  1.400000e+01  2.500000e+00   
max    2.650000e+02  2.650000e+02  4.000000e+00  1.238000e+03  7.000000e+00   

            mta_tax    tip_amount  tolls_amount  improvement_surcharge  \
count  1.048575e+06  1.048575e+06  1.048575e+06           1.048575e+06   
mean   4.928745e-01  2.086084e+00  3.829465e-01           2.978193e-01   
std    7.170903e-02  2.880471e+00  1.922991e+00           3.533964e-02   
min   -5.000000e-01 -7.000000e+00 -3.000000e+01          -3.000000e-01   
25%    5.000000e-01  0.000000e+00  0.000000e+00           3.000000e-01   
50%    5.000000e-01  1.750000e+00  0.000000e+00           3.000000e-01   
75%    5.000000e-01  2.750000e+00  0.000000e+00           3.000000e-01   
max    3.300000e+00  4.500000e+02  9.105000e+02           3.000000e-01   

       total_amount  congestion_surcharge  
count  1.048575e+06          1.048575e+06  
mean   1.880458e+01          2.250534e+00  
std    1.579603e+01          7.707165e-01  
min   -1.242300e+03         -2.500000e+00  
25%    1.080000e+01          2.500000e+00  
50%    1.400000e+01          2.500000e+00  
75%    1.956000e+01          2.500000e+00  
max    1.242300e+03          2.750000e+00

# Description of Dataset
## 1. Domain of the Dataset:
My dataset belongs to the transportation domain, specifically taxi trip records. It includes trip details such as pickup and drop-off times, fare amounts, distance, and payment details.

## 2. Target Variable:
The target variable from my dataset depends on the objective of analysis. Some possible targets from my dataset can be:

a. Total fare amount (total_amount): If predicting the fare amount is the goal, then this can be a target.

b. Trip distance (trip_distance): If analyzing trip distances, then this could be another target.

c. Payment type (payment_type): If classifying different payment methods, then this could be also a target.

### But the one I find most relevant is predicting the fare of the ride.

## 3. Number of Features and Records:
Features: The dataset that I chose contains 18 columns (features).

Records: Talking about the sample of my dataset, it contains 1048575 (approx 1.05 Million) records.

## 4. Type of Problem (Regression or Classification?):
As I am interested in predicting total fare amount, the problem is Regression as it has continuous output.